BHoM_Engine/Create/Type/EngineType.cs

```diff
@@ -49,14 +49,34 @@ public static Type EngineType(string name, bool silent = false, bool takeFirstIf
                 return null;
             }
 
-            // Try finding any types based on the assembly qualified name
+            // Try finding any types based on the assembly qualified name. This checks both assembly name as well as version
             List<Type> types = Global.EngineTypeList.Where(x => x.AssemblyQualifiedName == name).ToList();
             
             // If not found, look also based on unqualified name
             if (types.Count == 0)
             {
-                string unQualifiedName = name.Contains(",") ? name.Split(',').First() : name;
+                string unQualifiedName, assemblyName;
+                if (name.Contains(","))
+                {
+                    string[] split = name.Split(',');
+                    unQualifiedName = split[0];
+                    assemblyName = split[1].Trim(); //Assembly name is second part. Trim to remove whitespace
+                }
+                else
+                { 
+                    unQualifiedName = name;
+                    assemblyName = null;
+                }
+
                 types = Global.EngineTypeList.Where(x => x.FullName == unQualifiedName).ToList();
+
+                //If more than one type found, and assembly name was found, try use it to further filter
+                if (types.Count > 1 && assemblyName != null)
+                {
+                    List<Type> assemblyMatches = types.Where(x => x.Assembly.FullName.Split(',')[0] == assemblyName).ToList();  //Assembly fullname contains both name and version. Split removes the verison
```

Comment:
You could use the native members of `Assembly` class to get the name
```suggestion
                    List<Type> assemblyMatches = types.Where(x => x.Assembly.GetName().Name == assemblyName).ToList();  //Assembly fullname contains both name and version. Split removes the verison
```

Reply:
Ahh, great. Only looked for the Name on the assembly - missed the GetName(). Thanks. I commited separately to get rid of the comment that does not make sense any longer, rather than commiting your suggestion.